core/src/main/java/org/infinispan/partitionhandling/impl/PartitionHandlingManager.java

```diff
@@ -27,6 +27,14 @@
 
    void checkBulkRead();
 
+   /**
+    * Unlike in {@link #checkRead(Object)}, here we ignore the availability status and we only fail the operation if
+    * _all_ owners have left the cluster.
+    *
+    * @throws org.infinispan.partitionhandling.AvailabilityException if all owners of the keys are missing
+    */
+   void checkForMissingKeys(Collection<?> expected, Collection<?> actual);
```

Comment:
Hmm looking back at it I'm not too sure, it was a long time ago I wrote this part of the code. Maybe it's worth me removing this refactor and changing the behaviour in a separate PR?